Sources/TuistGenerator/Generator/ProjectFileElements.swift

```diff
@@ -619,6 +625,24 @@ class ProjectFileElements {
         }
     }
 
+    func variantGroup(containing localizedFile: AbsolutePath) -> (path: AbsolutePath, group: PBXVariantGroup)? {
+        let variantGroupPath = localizedFile.parentDirectory.parentDirectory.appending(
+            component: localizedFile.basename
+        )
+
+        for (elementPath, element) in elements {
```

Comment:
this can somewhat be expensive as we'd be hitting this code path several times throughout the generation and will be noticeable in larger projects 😞

An option could be to key variant groups differently in the elements dictionary, for example instead of `elements["/Path/Resources/App.xib"] = variantGroup` we could drop the extension and key it as `elements["/Path/Resources/App"] = variantGroup`. This would allow us to find the matching the variant group regardless of extension without needing to iterate over every single element. The one downside to this is will probably get confusing fairly fast (the logic is already not that straight forward 😅)

Along the same lines, we could possibly include multiple keys pointing to the same variant group, and have a dedicated variant groups dictionary to do the special lookup

- `variantGroups["/Path/Resources/App"] = variantGroup`
- `elements["/Path/Resources/Base.lproj/App.xib"] = variantGroup`
- `elements["/Path/Resources/en.lproj/App.strings"] = variantGroup`

This with a few helper methods may help us abstract away the details of how the variant group paths worth, especially from the build phase generator which has to sadly duplicate this logic today :( 

e.g.

```swift
// Build Phase Generator

// ...
if isLocalized {
   guard let (group, path) = fileElements.localizedElementGroup(path: buildFilePath) else {
            throw BuildPhaseGenerationError.missingFileReference(buildFilePath)
     }
     element = (group, path)
} 
```

Reply:
> An option could be to key variant groups differently in the elements dictionary, for example instead of `elements["/Path/Resources/App.xib"] = variantGroup` we could drop the extension and key it as `elements["/Path/Resources/App"] = variantGroup`.

The problem I see with this approach is that we can have more than one group sharing the same base name, i.e. if an app has both `App.strings` and `App.stringsdict` files we want to have to groups, one for each file type.

 > Along the same lines, we could possibly include multiple keys pointing to the same variant group, and have a dedicated variant groups dictionary to do the special lookup.

Thanks for this suggestion. I will go ahead and implement it that way 👍

Reply:
It turned out we don't even need a `variantGroups` property. The trick was simply to not loop through `elements` but the possible group extensions instead 😅